### PR TITLE
fwupd: 1.2.3 -> 1.2.8

### PR DESCRIPTION
**Pull request by dtzWill**
* use C.UTF-8 instead of en_US
* refresh sysconfdir_install patch
  * extend to vendor-directory.conf

###### Motivation for this change

Not sure how to link release notes reasonably,
they're in the tarballs IIRC.

Release tags:

https://github.com/hughsie/fwupd/releases/tag/1.2.5
https://github.com/hughsie/fwupd/releases/tag/1.2.4

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
@GrahamcOfBorg test fwupd

File: pkgs/os-specific/linux/firmware/fwupd/add-option-for-installation-sysconfdir.patch [hunk 1]
```diff
@@ -1,36 +1,8 @@
-From 44887227f7f617cbf84713ec45685cb4999039ff Mon Sep 17 00:00:00 2001
-From: Jan Tojnar <jtojnar@gmail.com>
-Date: Tue, 30 Oct 2018 22:26:30 +0100
-Subject: [PATCH] build: Add option for installation sysconfdir
```
Comment (worldofpeace) [inline pkgs/os-specific/linux/firmware/fwupd/add-option-for-installation-sysconfdir.patch]: Don't remove Jan's great commit msg :rofl:

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 2]
```diff
@@ -37,7 +37,7 @@ in stdenv.mkDerivation rec {
     bash-completion cairo freetype fontconfig pango
   ];
 
-  LC_ALL = "en_US.UTF-8"; # For po/make-images
+  LC_ALL = "C.UTF-8"; # For po/make-images
```
Comment (worldofpeace) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix]: Do we actually have to set this?

Comment (dtzWill) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix]: Not sure! :innocent:

File: pkgs/os-specific/linux/firmware/fwupd/add-option-for-installation-sysconfdir.patch (hunk 1 above)
Comment (dtzWill) [reply pkgs/os-specific/linux/firmware/fwupd/add-option-for-installation-sysconfdir.patch]: Just an artifact of how I regenerated it, haha, we can keep it for sure sorry :).

File: pkgs/os-specific/linux/firmware/fwupd/default.nix (hunk 2 above)
Comment (worldofpeace) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix]: Seemed to be fine https://github.com/NixOS/nixpkgs/pull/56390/commits/7e2fa85b62a3296205253d03a925aa3780e36f0a

**Comment by dtzWill** [comment]
How's this looking? I think w/the fixup commit this is good?

**Comment by worldofpeace** [comment]
Probably should run the tests, IIRC they timeout with ofborg.

**Comment by worldofpeace** [comment]
Did https://github.com/NixOS/nixpkgs/pull/56390/commits/e3b922acd75ba47c9f0a878350fb60f5bfb6e74a but mostly with the `fix-paths.patch`.

The tests succeed  for me locally.
##### Yet to look at [org.freedesktop.fwupd.metainfo.xml](https://github.com/hughsie/fwupd/blob/master/data/org.freedesktop.fwupd.metainfo.xml) in detail, but this really needs to be merged soon

**Comment by jtojnar** [comment]
> ##### Yet to look at [org.freedesktop.fwupd.metainfo.xml](https://github.com/hughsie/fwupd/blob/master/data/org.freedesktop.fwupd.metainfo.xml) [](/hughsie/fwupd/blob/HEAD@{2019-04-30T18:57:49Z}/data/org.freedesktop.fwupd.metainfo.xml) in detail, but this really needs to be merged soon

I just scanned it quickly and found the following possible integration issues:

> * Allow the fwupdmgr tool to modify the daemon config
> * Use Plymouth when updating offline firmware

**Comment by worldofpeace** [comment]
> Use Plymouth when updating offline firmware

It appears to only call into `plymouth` if it's installed (`g_find_program_in_path`).

**Comment by mweinelt** [comment]
This pull has become security critical for users of Logitech Unifying Devices, since fwupd before v1.2.7 has a bug, that prevents updating of the dongle.

A backport to NixOS 19.03 would therefore be much appreciated.

https://www.heise.de/ct/artikel/Logitech-keyboards-and-mice-vulnerable-to-extensive-cyber-attacks-4464533.html
https://www.cvedetails.com/cve/CVE-2019-13052/

**Comment by worldofpeace** [comment]
> This pull has become security critical for users of Logitech Unifying Devices, since fwupd before v1.2.7 has a bug, that prevents updating of the dongle.
> 
> [hughsie/fwupd#1139](https://github.com/hughsie/fwupd/issues/1139)
> [hughsie/fwupd@b62c3a4](https://github.com/hughsie/fwupd/commit/b62c3a4fae1104c4a1052adc7b3196afa8c33307)
> 
> A backport to NixOS 19.03 would therefore be much appreciated.
> 
> https://www.heise.de/ct/artikel/Logitech-keyboards-and-mice-vulnerable-to-extensive-cyber-attacks-4464533.html
> https://www.cvedetails.com/cve/CVE-2019-13052/



> This pull has become security critical for users of Logitech Unifying Devices, since fwupd before v1.2.7 has a bug, that prevents updating of the dongle.
> 
> [hughsie/fwupd#1139](https://github.com/hughsie/fwupd/issues/1139)
> [hughsie/fwupd@b62c3a4](https://github.com/hughsie/fwupd/commit/b62c3a4fae1104c4a1052adc7b3196afa8c33307)
> 
> A backport to NixOS 19.03 would therefore be much appreciated.
> 
> https://www.heise.de/ct/artikel/Logitech-keyboards-and-mice-vulnerable-to-extensive-cyber-attacks-4464533.html
> https://www.cvedetails.com/cve/CVE-2019-13052/

In that case it would be good to open an issue describing this.
For the case of `19.03` we probably can't backport this pr because I'm guessing
it's backwards compatability is low.

**Comment by worldofpeace** [comment]
*Squished* everything together.

After a brief checking I'm going to merge this as is. No point in reuseing the same PR
for the next bump, It'll never get merged :rofl:

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test fwupd

**Comment by worldofpeace** [comment]
@matthewbauer I think just the one commit needs a backport. (see https://github.com/NixOS/nixpkgs/issues/64631)

**Review by worldofpeace — APPROVED** [review]
Tests pass locally for me. Files in `filesInstalledToEtc` match after inspecting a `$ tree` output.

**Comment by matthewbauer** [comment]
Yeah probably just the version update

**Pull request merged**